weblab/entities/views.py

```diff
@@ -778,6 +780,54 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class TransferView(LoginRequiredMixin, UserFormKwargsMixin, UserPassesTestMixin,
+                   FormMixin, EntityTypeMixin, DetailView):
+    template_name = 'entities/entity_transfer_ownership.html'
+    context_object_name = 'entity'
+    form_class = EntityTransferForm
+
+    def _get_object(self):
+        if not hasattr(self, 'object'):
+            self.object = self.get_object()
+        return self.object
+
+    def test_func(self):
+        return self._get_object().is_managed_by(self.request.user)
+
+    def post(self, request, *args, **kwargs):
+        """Check the form and possibly rename the entity.
+
+        Called by Django when a form is submitted.
+        """
+        form = self.get_form()
+
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            entity = self.get_object()
+            user = User.objects.filter(email=email).first()
+
+            if user is None:
+                form.add_error(None, "Unknown user")
+                return self.form_invalid(form)
```

Comment:
It might be nice to do this in the same was as for `EntityCollaboratorForm`, i.e. add the same method `clean_email` to the form. It sets `self.cleaned_data['user']` if the user exists, or marks the form invalid if not. Then here we can just say `user = form.cleaned_data['user']`.

Reply:
thanks, I've put this in, it works for valid email addresses but for invalid ones I get an error 

_The view entities.views.TransferView didn't return an HttpResponse object. It returned None instead._

When you say the clean_email method "marks the form invalid if not" do I need to do this explicitly or is  `raise ValidationError('User not found')` sufficient?

Reply:
The `raise` does that. I think the error is because you don't have an else branch in your view on the `if form.is_valid()`?

Reply:
yes that did the trick thanks, I put `return self.form_invalid(form)` in the else branch and the page now works as expected